src/ucp/core/ucp_ep.c

```diff
@@ -209,19 +218,32 @@ ucs_status_t ucp_ep_create(ucp_worker_h worker,
     }
 
     /* initialize transport endpoints */
-    status = ucp_wireup_init_lanes(ep, address_count, address_list, addr_indices);
+    status = ucp_wireup_init_lanes(ep, params, address_count, address_list,
+                                   addr_indices);
     if (status != UCS_OK) {
         goto err_destroy_ep;
     }
 
-    if (params->field_mask & UCP_EP_PARAM_FIELD_ERR_HANDLING_MODE) {
+    /* Setup error handler */
+    if (params->field_mask & UCP_EP_PARAM_FIELD_ERR_HANDLER) {
+        hash_it = kh_put(ucp_ep_errh_hash, &worker->ep_errh_hash, (uintptr_t)ep,
+                         &hash_extra_status);
+        if (ucs_unlikely(hash_it == kh_end(&worker->ep_errh_hash))) {
+            ucs_error("Hash failed on setup error handler of endpoint %p with status %d ",
+                      ep, hash_extra_status);
+            status = UCS_ERR_NO_RESOURCE;
+            goto err_destroy_ep;
+        }
+        kh_value(&worker->ep_errh_hash, hash_it) = params->err_handler;
+
         ucp_ep_config(ep)->err_mode = params->err_mode;
+    }
+
+    if (params->field_mask & UCP_EP_PARAM_FIELD_ERR_HANDLING_MODE) {
```

Comment:
need to move this code to ucp_ep_config_init() to set rndv_thresh based on key->err_mode